cnf-certification-test/accesscontrol/rbac/automount.go

```diff
@@ -0,0 +1,65 @@
+package rbac
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type AutomountToken struct {
+	serviceAccountName string
+	podNamespace       string
+	ClientHolder       *clientsholder.ClientsHolder
+}
+
+type ServiceAccountTokenStatus struct {
+	Name       string
+	TokenSet   bool
+	TokenValue bool
+}
+
+func NewAutomountTester(serviceAccountName, podNamespace string, ch *clientsholder.ClientsHolder) *AutomountToken {
+	return &AutomountToken{
+		serviceAccountName: serviceAccountName,
+		podNamespace:       podNamespace,
+		ClientHolder:       ch,
+	}
+}
+
+func (at *AutomountToken) AutomountServiceAccountSetOnSA() (ServiceAccountTokenStatus, error) {
+	// Create a default SATS struct.
+	saStatus := ServiceAccountTokenStatus{
+		TokenSet:   false,
+		TokenValue: false,
+	}
+
+	// Collect all of the service accounts that live in the target namespace.
+	saList, saErr := at.ClientHolder.K8sClient.CoreV1().ServiceAccounts(at.podNamespace).List(context.TODO(), v1.ListOptions{})
```

Comment:
do you need all of them ? I think you can use Get and provide the one you need. The whole list is not needed

Reply:
Good point.  I'll check that out.